---
title: "ecoli.all"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## R Markdown
E.coli  -- Analzye all data simultaneously

data was aligned to ecoli /data/strickt2/SEQreference/new_ecoli/uti89.final.fa with bwa. Reads were counted using htseq, with /data/strickt2/SEQreference/new_ecoli/Escherichia_coli_uti89.ASM1326v1.31.gff3 as a reference gtf. 
Below, data is analyzed with DESeq2, using a model with terms for strain, time, and an interaction between the two. 

```{r differential expression all genes, echo=TRUE, message=FALSE, warning=FALSE}
library("DESeq2")
library("ggplot2")
library("qvalue")
library("pheatmap")
library("RColorBrewer")
library("sva")
library("PoiClaClu")
library("sva")
library("ReportingTools")


directory <- "~/ecoli/data/count/"
sampleFiles <- grep("count$",list.files(directory),value=TRUE)

sampleCondition <-read.delim(file ="~/ecoli/ecoli.1.3.sample.table.txt", header =T, as.is =T, quote ="")
sampleTable <- data.frame(sampleName = sampleFiles,fileName = sampleFiles,condition = sampleCondition)
count = read.delim(file ="~/ecoli/data/count/count.matrix.txt", header =T, as.is =T, quote ="", row.names = 1)
count1=count[,-43]

ddsHTSeq <-  DESeqDataSetFromMatrix(countData = count1,colData = sampleCondition, design=~Batch + Strain + Time_point + Strain:Time_point)

dds <- ddsHTSeq[ rowSums(counts(ddsHTSeq)) > 1, ]
dds <- DESeq(dds)

#dat  <- counts(dds, normalized = TRUE)
#idx  <- rowMeans(dat) > 1
#dat  <- dat[idx, ]
#mod  <- model.matrix(~Batch + Strain + Time_point + Strain:Time_point, colData(dds))
#mod0 <- model.matrix(~   1, colData(dds))
#svseq <- svaseq(dat, mod, mod0)

#ddssva <- dds
#ddssva$SV1 <- svseq$sv[,1]
#ddssva$SV2 <- svseq$sv[,2]
#ddssva$SV3 <- svseq$sv[,3]
#ddssva$SV4 <- svseq$sv[,4]
#design(ddssva) <- ~ SV1+SV2 +SV3 +SV4+ Batch + Strain + Time_point + Strain:Time_point

#ddssva <-DESeq(ddssva)

ddsTC <- DESeq(dds, test="LRT", reduced = ~Batch + Strain + Time_point)
resTC <- results(ddsTC)
resTC$symbol <- mcols(ddsTC)$symbol
head(resTC[order(resTC$pvalue),])
sum(resTC$padj < 0.1, na.rm=TRUE)

res15 = results(ddsTC, name = "StraindpmrB.Time_point60m")
summary(res15)
head(res15[order(res15$pvalue),])


resultsNames(ddsTC)
```

```{r subset chunk 1}
wt = c(1:6)
dpmrA = c(wt,7:12)
dpmrB =c(wt,13:18)
dqseB =c(wt,19:24)
dqseBdpmrA = c(wt,25:30)
dqseC = c(wt,31:36)
dqseCdpmrA = c(wt,37:42)

gene = "dpmrA"
  
ddsHTSeq <-  DESeqDataSetFromMatrix(countData = count1[,dpmrA],colData = sampleCondition[dpmrA,], design=~Strain + Time_point +Batch)
dds <- ddsHTSeq[ rowSums(counts(ddsHTSeq)) > 1, ]
dds <- DESeq(dds)
resultsNames(dds)
names = resultsNames(dds)
res <- results(dds, contrast=c("Strain",gene,"WT"))
summary(res)
head(res[order(res$pvalue),])
resSig <- subset(res, padj < 0.1)
head(resSig[ order(resSig$log2FoldChange), ])
resOrderedDF <- as.data.frame(resSig)
nam = c("gene", colnames(resOrderedDF))
res2 = cbind(rownames(resOrderedDF), resOrderedDF)
colnames(res2) = nam
file1 = paste(gene, "strain.results.csv",  sep = ".")
name1 = paste(gene, "strain report", sep = " ")
name2 = paste(gene, "vs wild-type", sep = " ")
write.csv(res2, file = file1, row.names = T)
htmlRep <- HTMLReport(shortName=name1, title=name2,
                      reportDirectory="./report")
publish(res2, htmlRep)
url <- finish(htmlRep)
browseURL(url)



res <- results(dds, contrast=c("Time_point","0m","15m"))
summary(res)
head(res[order(res$pvalue),])
resSig <- subset(res, padj < 0.1)
head(resSig[ order(resSig$log2FoldChange), ])
resOrderedDF <- as.data.frame(resSig)
nam = c("gene", colnames(resOrderedDF))
res2 = cbind(rownames(resOrderedDF), resOrderedDF)
file1 = paste(gene, "time015.results.csv",  sep = ".")
name1 = paste(gene, "time 015 report", sep = " ")
name2 = paste(gene, "0 15 report", sep = " ")
write.csv(res2, file = file1, row.names = T)
htmlRep <- HTMLReport(shortName=name1, title=name2,
                      reportDirectory="./report")
publish(res2, htmlRep)
url <- finish(htmlRep)
browseURL(url)




res <- results(dds, contrast=c("Time_point","0m","60m"))
summary(res)
head(res[order(res$pvalue),])
resSig <- subset(res, padj < 0.1)
head(resSig[ order(resSig$log2FoldChange), ])
resOrderedDF <- as.data.frame(resSig)
nam = c("gene", colnames(resOrderedDF))
res2 = cbind(rownames(resOrderedDF), resOrderedDF)
colnames(res2) = nam

file1 = paste(gene, "time060.results.csv",  sep = ".")
name1 = paste(gene, "time060 report", sep = " ")
name2 = paste(gene, "0m 60m report", sep = " ")
write.csv(res2, file = file1, row.names = T)
htmlRep <- HTMLReport(shortName=name1, title=name2,
                      reportDirectory="./report")
publish(res2, htmlRep)
url <- finish(htmlRep)
browseURL(url)



res <- results(dds, contrast=c("Time_point","15m","60m"))
summary(res)
head(res[order(res$pvalue),])
resSig <- subset(res, padj < 0.1)
head(resSig[ order(resSig$log2FoldChange), ])
resOrderedDF <- as.data.frame(resSig)
nam = c("gene", colnames(resOrderedDF))
res2 = cbind(rownames(resOrderedDF), resOrderedDF)
colnames(res2) = nam

file1 = paste(gene, "time1560.results.csv",  sep = ".")
name1 = paste(gene, "time1560 report", sep = " ")
name2 = paste(gene, "15m 60m report", sep = " ")
write.csv(res2, file = file1, row.names = T)
htmlRep <- HTMLReport(shortName=name1, title=name2,
                      reportDirectory="./report")
publish(res2, htmlRep)
url <- finish(htmlRep)
browseURL(url)



```

```{r subset chunk 1}
wt = c(1:6)
dpmrA = c(wt,7:12)
dpmrB =c(wt,13:18)
dqseB =c(wt,19:24)
dqseBdpmrA = c(wt,25:30)
dqseC = c(wt,31:36)
dqseCdpmrA = c(wt,37:42)

gene = "dpmrB"
  
ddsHTSeq <-  DESeqDataSetFromMatrix(countData = count1[,dpmrB],colData = sampleCondition[dpmrB,], design=~Strain + Time_point +Batch)
dds <- ddsHTSeq[ rowSums(counts(ddsHTSeq)) > 1, ]
dds <- DESeq(dds)
resultsNames(dds)
names = resultsNames(dds)
res <- results(dds, contrast=c("Strain",gene,"WT"))
summary(res)
head(res[order(res$pvalue),])
resSig <- subset(res, padj < 0.1)
head(resSig[ order(resSig$log2FoldChange), ])
resOrderedDF <- as.data.frame(resSig)
nam = c("gene", colnames(resOrderedDF))
res2 = cbind(rownames(resOrderedDF), resOrderedDF)
colnames(res2) = nam
file1 = paste(gene, "strain.results.csv",  sep = ".")
name1 = paste(gene, "strain report", sep = " ")
name2 = paste(gene, "vs wild-type", sep = " ")
write.csv(res2, file = file1, row.names = T)
htmlRep <- HTMLReport(shortName=name1, title=name2,
                      reportDirectory="./report")
publish(res2, htmlRep)
url <- finish(htmlRep)
browseURL(url)



res <- results(dds, contrast=c("Time_point","0m","15m"))
summary(res)
head(res[order(res$pvalue),])
resSig <- subset(res, padj < 0.1)
head(resSig[ order(resSig$log2FoldChange), ])
resOrderedDF <- as.data.frame(resSig)
nam = c("gene", colnames(resOrderedDF))
res2 = cbind(rownames(resOrderedDF), resOrderedDF)
file1 = paste(gene, "time015.results.csv",  sep = ".")
name1 = paste(gene, "time 015 report", sep = " ")
name2 = paste(gene, "0 15 report", sep = " ")
write.csv(res2, file = file1, row.names = T)
htmlRep <- HTMLReport(shortName=name1, title=name2,
                      reportDirectory="./report")
publish(res2, htmlRep)
url <- finish(htmlRep)
browseURL(url)




res <- results(dds, contrast=c("Time_point","0m","60m"))
summary(res)
head(res[order(res$pvalue),])
resSig <- subset(res, padj < 0.1)
head(resSig[ order(resSig$log2FoldChange), ])
resOrderedDF <- as.data.frame(resSig)
nam = c("gene", colnames(resOrderedDF))
res2 = cbind(rownames(resOrderedDF), resOrderedDF)
colnames(res2) = nam

file1 = paste(gene, "time060.results.csv",  sep = ".")
name1 = paste(gene, "time060 report", sep = " ")
name2 = paste(gene, "0m 60m report", sep = " ")
write.csv(res2, file = file1, row.names = T)
htmlRep <- HTMLReport(shortName=name1, title=name2,
                      reportDirectory="./report")
publish(res2, htmlRep)
url <- finish(htmlRep)
browseURL(url)



res <- results(dds, contrast=c("Time_point","15m","60m"))
summary(res)
head(res[order(res$pvalue),])
resSig <- subset(res, padj < 0.1)
head(resSig[ order(resSig$log2FoldChange), ])
resOrderedDF <- as.data.frame(resSig)
nam = c("gene", colnames(resOrderedDF))
res2 = cbind(rownames(resOrderedDF), resOrderedDF)
colnames(res2) = nam

file1 = paste(gene, "time1560.results.csv",  sep = ".")
name1 = paste(gene, "time1560 report", sep = " ")
name2 = paste(gene, "15m 60m report", sep = " ")
write.csv(res2, file = file1, row.names = T)
htmlRep <- HTMLReport(shortName=name1, title=name2,
                      reportDirectory="./report")
publish(res2, htmlRep)
url <- finish(htmlRep)
browseURL(url)



```

```{r subset chunk 1}
wt = c(1:6)
dpmrA = c(wt,7:12)
dpmrB =c(wt,13:18)
dqseB =c(wt,19:24)
dqseBdpmrA = c(wt,25:30)
dqseC = c(wt,31:36)
dqseCdpmrA = c(wt,37:42)

gene = "dqseB"
  
ddsHTSeq <-  DESeqDataSetFromMatrix(countData = count1[,dqseB],colData = sampleCondition[dqseB,], design=~Strain + Time_point +Batch)
dds <- ddsHTSeq[ rowSums(counts(ddsHTSeq)) > 1, ]
dds <- DESeq(dds)
resultsNames(dds)
names = resultsNames(dds)
res <- results(dds, contrast=c("Strain",gene,"WT"))
summary(res)
head(res[order(res$pvalue),])
resSig <- subset(res, padj < 0.1)
head(resSig[ order(resSig$log2FoldChange), ])
resOrderedDF <- as.data.frame(resSig)
nam = c("gene", colnames(resOrderedDF))
res2 = cbind(rownames(resOrderedDF), resOrderedDF)
colnames(res2) = nam
file1 = paste(gene, "strain.results.csv",  sep = ".")
name1 = paste(gene, "strain report", sep = " ")
name2 = paste(gene, "vs wild-type", sep = " ")
write.csv(res2, file = file1, row.names = T)
htmlRep <- HTMLReport(shortName=name1, title=name2,
                      reportDirectory="./report")
publish(res2, htmlRep)
url <- finish(htmlRep)
browseURL(url)



res <- results(dds, contrast=c("Time_point","0m","15m"))
summary(res)
head(res[order(res$pvalue),])
resSig <- subset(res, padj < 0.1)
head(resSig[ order(resSig$log2FoldChange), ])
resOrderedDF <- as.data.frame(resSig)
nam = c("gene", colnames(resOrderedDF))
res2 = cbind(rownames(resOrderedDF), resOrderedDF)
file1 = paste(gene, "time015.results.csv",  sep = ".")
name1 = paste(gene, "time 015 report", sep = " ")
name2 = paste(gene, "0 15 report", sep = " ")
write.csv(res2, file = file1, row.names = T)
htmlRep <- HTMLReport(shortName=name1, title=name2,
                      reportDirectory="./report")
publish(res2, htmlRep)
url <- finish(htmlRep)
browseURL(url)




res <- results(dds, contrast=c("Time_point","0m","60m"))
summary(res)
head(res[order(res$pvalue),])
resSig <- subset(res, padj < 0.1)
head(resSig[ order(resSig$log2FoldChange), ])
resOrderedDF <- as.data.frame(resSig)
nam = c("gene", colnames(resOrderedDF))
res2 = cbind(rownames(resOrderedDF), resOrderedDF)
colnames(res2) = nam

file1 = paste(gene, "time060.results.csv",  sep = ".")
name1 = paste(gene, "time060 report", sep = " ")
name2 = paste(gene, "0m 60m report", sep = " ")
write.csv(res2, file = file1, row.names = T)
htmlRep <- HTMLReport(shortName=name1, title=name2,
                      reportDirectory="./report")
publish(res2, htmlRep)
url <- finish(htmlRep)
browseURL(url)



res <- results(dds, contrast=c("Time_point","15m","60m"))
summary(res)
head(res[order(res$pvalue),])
resSig <- subset(res, padj < 0.1)
head(resSig[ order(resSig$log2FoldChange), ])
resOrderedDF <- as.data.frame(resSig)
nam = c("gene", colnames(resOrderedDF))
res2 = cbind(rownames(resOrderedDF), resOrderedDF)
colnames(res2) = nam

file1 = paste(gene, "time1560.results.csv",  sep = ".")
name1 = paste(gene, "time1560 report", sep = " ")
name2 = paste(gene, "15m 60m report", sep = " ")
write.csv(res2, file = file1, row.names = T)
htmlRep <- HTMLReport(shortName=name1, title=name2,
                      reportDirectory="./report")
publish(res2, htmlRep)
url <- finish(htmlRep)
browseURL(url)



```

```{r subset chunk 1}
wt = c(1:6)
dpmrA = c(wt,7:12)
dpmrB =c(wt,13:18)
dqseB =c(wt,19:24)
dqseBdpmrA = c(wt,25:30)
dqseC = c(wt,31:36)
dqseCdpmrA = c(wt,37:42)

gene = "dqseBdpmrA"
  
ddsHTSeq <-  DESeqDataSetFromMatrix(countData = count1[,dqseBdpmrA],colData = sampleCondition[dqseBdpmrA,], design=~Strain + Time_point +Batch)
dds <- ddsHTSeq[ rowSums(counts(ddsHTSeq)) > 1, ]
dds <- DESeq(dds)
resultsNames(dds)
names = resultsNames(dds)
res <- results(dds, contrast=c("Strain",gene,"WT"))
summary(res)
head(res[order(res$pvalue),])
resSig <- subset(res, padj < 0.1)
head(resSig[ order(resSig$log2FoldChange), ])
resOrderedDF <- as.data.frame(resSig)
nam = c("gene", colnames(resOrderedDF))
res2 = cbind(rownames(resOrderedDF), resOrderedDF)
colnames(res2) = nam
file1 = paste(gene, "strain.results.csv",  sep = ".")
name1 = paste(gene, "strain report", sep = " ")
name2 = paste(gene, "vs wild-type", sep = " ")
write.csv(res2, file = file1, row.names = T)
htmlRep <- HTMLReport(shortName=name1, title=name2,
                      reportDirectory="./report")
publish(res2, htmlRep)
url <- finish(htmlRep)
browseURL(url)



res <- results(dds, contrast=c("Time_point","0m","15m"))
summary(res)
head(res[order(res$pvalue),])
resSig <- subset(res, padj < 0.1)
head(resSig[ order(resSig$log2FoldChange), ])
resOrderedDF <- as.data.frame(resSig)
nam = c("gene", colnames(resOrderedDF))
res2 = cbind(rownames(resOrderedDF), resOrderedDF)
file1 = paste(gene, "time015.results.csv",  sep = ".")
name1 = paste(gene, "time 015 report", sep = " ")
name2 = paste(gene, "0 15 report", sep = " ")
write.csv(res2, file = file1, row.names = T)
htmlRep <- HTMLReport(shortName=name1, title=name2,
                      reportDirectory="./report")
publish(res2, htmlRep)
url <- finish(htmlRep)
browseURL(url)




res <- results(dds, contrast=c("Time_point","0m","60m"))
summary(res)
head(res[order(res$pvalue),])
resSig <- subset(res, padj < 0.1)
head(resSig[ order(resSig$log2FoldChange), ])
resOrderedDF <- as.data.frame(resSig)
nam = c("gene", colnames(resOrderedDF))
res2 = cbind(rownames(resOrderedDF), resOrderedDF)
colnames(res2) = nam

file1 = paste(gene, "time060.results.csv",  sep = ".")
name1 = paste(gene, "time060 report", sep = " ")
name2 = paste(gene, "0m 60m report", sep = " ")
write.csv(res2, file = file1, row.names = T)
htmlRep <- HTMLReport(shortName=name1, title=name2,
                      reportDirectory="./report")
publish(res2, htmlRep)
url <- finish(htmlRep)
browseURL(url)



res <- results(dds, contrast=c("Time_point","15m","60m"))
summary(res)
head(res[order(res$pvalue),])
resSig <- subset(res, padj < 0.1)
head(resSig[ order(resSig$log2FoldChange), ])
resOrderedDF <- as.data.frame(resSig)
nam = c("gene", colnames(resOrderedDF))
res2 = cbind(rownames(resOrderedDF), resOrderedDF)
colnames(res2) = nam

file1 = paste(gene, "time1560.results.csv",  sep = ".")
name1 = paste(gene, "time1560 report", sep = " ")
name2 = paste(gene, "15m 60m report", sep = " ")
write.csv(res2, file = file1, row.names = T)
htmlRep <- HTMLReport(shortName=name1, title=name2,
                      reportDirectory="./report")
publish(res2, htmlRep)
url <- finish(htmlRep)
browseURL(url)



```

```{r subset chunk 1}
wt = c(1:6)
dpmrA = c(wt,7:12)
dpmrB =c(wt,13:18)
dqseB =c(wt,19:24)
dqseBdpmrA = c(wt,25:30)
dqseC = c(wt,31:36)
dqseCdpmrA = c(wt,37:42)

gene = "dqseC"
  
ddsHTSeq <-  DESeqDataSetFromMatrix(countData = count1[,dqseC],colData = sampleCondition[dqseC,], design=~Strain + Time_point +Batch)
dds <- ddsHTSeq[ rowSums(counts(ddsHTSeq)) > 1, ]
dds <- DESeq(dds)
resultsNames(dds)
names = resultsNames(dds)
res <- results(dds, contrast=c("Strain",gene,"WT"))
summary(res)
head(res[order(res$pvalue),])
resSig <- subset(res, padj < 0.1)
head(resSig[ order(resSig$log2FoldChange), ])
resOrderedDF <- as.data.frame(resSig)
nam = c("gene", colnames(resOrderedDF))
res2 = cbind(rownames(resOrderedDF), resOrderedDF)
colnames(res2) = nam
file1 = paste(gene, "strain.results.csv",  sep = ".")
name1 = paste(gene, "strain report", sep = " ")
name2 = paste(gene, "vs wild-type", sep = " ")
write.csv(res2, file = file1, row.names = T)
htmlRep <- HTMLReport(shortName=name1, title=name2,
                      reportDirectory="./report")
publish(res2, htmlRep)
url <- finish(htmlRep)
browseURL(url)



res <- results(dds, contrast=c("Time_point","0m","15m"))
summary(res)
head(res[order(res$pvalue),])
resSig <- subset(res, padj < 0.1)
head(resSig[ order(resSig$log2FoldChange), ])
resOrderedDF <- as.data.frame(resSig)
nam = c("gene", colnames(resOrderedDF))
res2 = cbind(rownames(resOrderedDF), resOrderedDF)
file1 = paste(gene, "time015.results.csv",  sep = ".")
name1 = paste(gene, "time 015 report", sep = " ")
name2 = paste(gene, "0 15 report", sep = " ")
write.csv(res2, file = file1, row.names = T)
htmlRep <- HTMLReport(shortName=name1, title=name2,
                      reportDirectory="./report")
publish(res2, htmlRep)
url <- finish(htmlRep)
browseURL(url)




res <- results(dds, contrast=c("Time_point","0m","60m"))
summary(res)
head(res[order(res$pvalue),])
resSig <- subset(res, padj < 0.1)
head(resSig[ order(resSig$log2FoldChange), ])
resOrderedDF <- as.data.frame(resSig)
nam = c("gene", colnames(resOrderedDF))
res2 = cbind(rownames(resOrderedDF), resOrderedDF)
colnames(res2) = nam

file1 = paste(gene, "time060.results.csv",  sep = ".")
name1 = paste(gene, "time060 report", sep = " ")
name2 = paste(gene, "0m 60m report", sep = " ")
write.csv(res2, file = file1, row.names = T)
htmlRep <- HTMLReport(shortName=name1, title=name2,
                      reportDirectory="./report")
publish(res2, htmlRep)
url <- finish(htmlRep)
browseURL(url)



res <- results(dds, contrast=c("Time_point","15m","60m"))
summary(res)
head(res[order(res$pvalue),])
resSig <- subset(res, padj < 0.1)
head(resSig[ order(resSig$log2FoldChange), ])
resOrderedDF <- as.data.frame(resSig)
nam = c("gene", colnames(resOrderedDF))
res2 = cbind(rownames(resOrderedDF), resOrderedDF)
colnames(res2) = nam

file1 = paste(gene, "time1560.results.csv",  sep = ".")
name1 = paste(gene, "time1560 report", sep = " ")
name2 = paste(gene, "15m 60m report", sep = " ")
write.csv(res2, file = file1, row.names = T)
htmlRep <- HTMLReport(shortName=name1, title=name2,
                      reportDirectory="./report")
publish(res2, htmlRep)
url <- finish(htmlRep)
browseURL(url)



```
```{r subset chunk 1}
wt = c(1:6)
dpmrA = c(wt,7:12)
dpmrB =c(wt,13:18)
dqseB =c(wt,19:24)
dqseBdpmrA = c(wt,25:30)
dqseC = c(wt,31:36)
dqseCdpmrA = c(wt,37:42)

gene = "dqseCdpmrA"
  
ddsHTSeq <-  DESeqDataSetFromMatrix(countData = count1[,dqseCdpmrA],colData = sampleCondition[dqseCdpmrA,], design=~Strain + Time_point +Batch)
dds <- ddsHTSeq[ rowSums(counts(ddsHTSeq)) > 1, ]
dds <- DESeq(dds)
resultsNames(dds)
names = resultsNames(dds)
res <- results(dds, contrast=c("Strain",gene,"WT"))
summary(res)
head(res[order(res$pvalue),])
resSig <- subset(res, padj < 0.1)
head(resSig[ order(resSig$log2FoldChange), ])
resOrderedDF <- as.data.frame(resSig)
nam = c("gene", colnames(resOrderedDF))
res2 = cbind(rownames(resOrderedDF), resOrderedDF)
colnames(res2) = nam
file1 = paste(gene, "strain.results.csv",  sep = ".")
name1 = paste(gene, "strain report", sep = " ")
name2 = paste(gene, "vs wild-type", sep = " ")
write.csv(res2, file = file1, row.names = T)
htmlRep <- HTMLReport(shortName=name1, title=name2,
                      reportDirectory="./report")
publish(res2, htmlRep)
url <- finish(htmlRep)
browseURL(url)



res <- results(dds, contrast=c("Time_point","0m","15m"))
summary(res)
head(res[order(res$pvalue),])
resSig <- subset(res, padj < 0.1)
head(resSig[ order(resSig$log2FoldChange), ])
resOrderedDF <- as.data.frame(resSig)
nam = c("gene", colnames(resOrderedDF))
res2 = cbind(rownames(resOrderedDF), resOrderedDF)
file1 = paste(gene, "time015.results.csv",  sep = ".")
name1 = paste(gene, "time 015 report", sep = " ")
name2 = paste(gene, "0 15 report", sep = " ")
write.csv(res2, file = file1, row.names = T)
htmlRep <- HTMLReport(shortName=name1, title=name2,
                      reportDirectory="./report")
publish(res2, htmlRep)
url <- finish(htmlRep)
browseURL(url)




res <- results(dds, contrast=c("Time_point","0m","60m"))
summary(res)
head(res[order(res$pvalue),])
resSig <- subset(res, padj < 0.1)
head(resSig[ order(resSig$log2FoldChange), ])
resOrderedDF <- as.data.frame(resSig)
nam = c("gene", colnames(resOrderedDF))
res2 = cbind(rownames(resOrderedDF), resOrderedDF)
colnames(res2) = nam

file1 = paste(gene, "time060.results.csv",  sep = ".")
name1 = paste(gene, "time060 report", sep = " ")
name2 = paste(gene, "0m 60m report", sep = " ")
write.csv(res2, file = file1, row.names = T)
htmlRep <- HTMLReport(shortName=name1, title=name2,
                      reportDirectory="./report")
publish(res2, htmlRep)
url <- finish(htmlRep)
browseURL(url)



res <- results(dds, contrast=c("Time_point","15m","60m"))
summary(res)
head(res[order(res$pvalue),])
resSig <- subset(res, padj < 0.1)
head(resSig[ order(resSig$log2FoldChange), ])
resOrderedDF <- as.data.frame(resSig)
nam = c("gene", colnames(resOrderedDF))
res2 = cbind(rownames(resOrderedDF), resOrderedDF)
colnames(res2) = nam

file1 = paste(gene, "time1560.results.csv",  sep = ".")
name1 = paste(gene, "time1560 report", sep = " ")
name2 = paste(gene, "15m 60m report", sep = " ")
write.csv(res2, file = file1, row.names = T)
htmlRep <- HTMLReport(shortName=name1, title=name2,
                      reportDirectory="./report")
publish(res2, htmlRep)
url <- finish(htmlRep)
browseURL(url)



```

plots -- plots for all data 

```{r plots all data,  message=FALSE, warning=FALSE}
rld <- rlog(dds, blind=FALSE)

sampleDists <- dist( t( assay(rld) ) )
sampleDistMatrix <- as.matrix( sampleDists )
rownames(sampleDistMatrix) <-paste(sampleCondition[,1],sampleCondition[,2],sampleCondition[,4], sep =".") 
colnames(sampleDistMatrix) <- NULL
colors <- colorRampPalette( rev(brewer.pal(9, "Blues")) )(255)
pheatmap(sampleDistMatrix,
         clustering_distance_rows=sampleDists,
         clustering_distance_cols=sampleDists,
         col=colors)

poisd <- PoissonDistance(t(counts(dds)))

samplePoisDistMatrix <- as.matrix( poisd$dd )
rownames(samplePoisDistMatrix) <- paste(sampleCondition[,1],sampleCondition[,2],sampleCondition[,4], sep =".") 
colnames(samplePoisDistMatrix) <- NULL
pheatmap(samplePoisDistMatrix,
         clustering_distance_rows=poisd$dd,
         clustering_distance_cols=poisd$dd,
         col=colors)

plotPCA(rld, intgroup = c("Strain", "Time_point"))

pcaData <- plotPCA(rld, intgroup = c( "Strain", "Time_point"), returnData=TRUE)
percentVar <- round(100 * attr(pcaData, "percentVar"))

ggplot(pcaData, aes(PC1, PC2, color=condition.Strain, shape=condition.Time_point)) + geom_point(size=3) +
  xlab(paste0("PC1: ",percentVar[1],"% variance")) +
  ylab(paste0("PC2: ",percentVar[2],"% variance")) +
  coord_fixed()

mdsData <- data.frame(cmdscale(sampleDistMatrix))
mds <- cbind(mdsData, as.data.frame(colData(rld)))
ggplot(mds, aes(X1,X2,color=condition.Strain,shape=condition.Time_point)) + geom_point(size=3) +
  coord_fixed()

mdsPoisData <- data.frame(cmdscale(samplePoisDistMatrix))
mdsPois <- cbind(mdsPoisData, as.data.frame(colData(dds)))
ggplot(mdsPois, aes(X1,X2,color=condition.Strain,shape=condition.Time_point)) + geom_point(size=3) +
  coord_fixed()

qs <- c(0, quantile(resTC$baseMean[resTC$baseMean > 0], 0:6/6))
bins <- cut(resTC$baseMean, qs)
levels(bins) <- paste0("~",round(signif(.5*qs[-1] + .5*qs[-length(qs)],2)))
ratios <- tapply(resTC$pvalue, bins, function(p) mean(p < .05, na.rm=TRUE))
barplot(ratios, xlab="mean normalized count", ylab="ratio of small p values")

```

```{r short }

count.short = count1[,1:12]
sC.short = sampleCondition[1:12,]


ddsHTSeq <-  DESeqDataSetFromMatrix(countData = count.short,colData = sC.short, design=~Time_point + Strain + Strain:Time_point)
dds <- ddsHTSeq[ rowSums(counts(ddsHTSeq)) > 1, ]
dds <- DESeq(dds)
ddsTC <- DESeq(dds, test="LRT", reduced = ~Time_point+Strain)
resTC <- results(ddsTC)
resTC$symbol <- mcols(ddsTC)$symbol
head(resTC[order(resTC$pvalue),])
sum(resTC$padj < 0.1, na.rm=TRUE)


dds <- ddsHTSeq[ rowSums(counts(ddsHTSeq)) > 1, ]
dds <- DESeq(dds)
res <- results(dds)
res
summary(res)
res.05 <- results(dds, alpha = 0.05)
table(res.05$padj < 0.05)

res15 = results(dds, contrast = c("Time_point", "15m", "0m"))
summary(res15)

res.st = results(dds, contrast = c("Strain", "dpmrA", "WT"))
summary(res.st)



fiss <- plotCounts(ddsTC, which.min(resTC$padj), 
                   intgroup = c("Time_point","Strain"), returnData = TRUE)
ggplot(fiss,
  aes(x = as.numeric(Time_point), y = count, color = Strain, group = Strain)) + 
  geom_point() + geom_smooth(se = FALSE, method = "loess") + scale_y_log10()

```